File: java/prototype/Poster.java
package prototype;

public class Poster extends MarketingMaterial {
    public Poster(String layout, String content, String color) {
        super(layout, content, color);
    }

    @Override
    public Poster cloneMarketingMaterial() {
        return new Poster(this.getLayout(), this.getContent(), this.getColor());
    }
}

File: C#/Prototype/Poster.cs
namespace Prototype;

public sealed class Poster : MarketingMaterial
{
    public Poster(string layout, string content, string color) : base(layout, content, color)
    {
    }

    public override MarketingMaterial CloneMarketingMaterial()
        => new Poster(Layout, Content, Color);
}